### PR TITLE
feat: Wrap all call state hooks in useCallStateHooks()

### PR DESCRIPTION
**Pull request by oliverlaz**
### Overview

In order to improve the discoverability of our state-access hooks, all of them are now exposed through one hook-alike function: `useCallStateHooks()`;

Additionally, `useCallMetadata` is now deprecated and will be removed in a follow-up release.

#### Integration:
**Before**
```tsx
import { useLocalParticipant,  useRemoteParticipants } from '@stream-io/video-react-sdk';
// or: import { useLocalParticipant,  useRemoteParticipants } from '@stream-io/video-react-native-sdk';
// or: import { useLocalParticipant,  useRemoteParticipants } from '@stream-io/video-react-bindings';

export const MyComponent = () => {
  const localParticipant = useLocalParticipant();
  const remoteParticipants = useRemoteParticipants();
  ...
};
```

**After**
```tsx
import { useCallStateHooks } from '@stream-io/video-react-sdk';
// or: import { useCallStateHooks } from '@stream-io/video-react-native-sdk';
// or: import { useCallStateHooks } from '@stream-io/video-react-bindings';

export const MyComponent = () => {
  const { useLocalParticipant, useRemoteParticipants } = useCallStateHooks();
  const localParticipant = useLocalParticipant();
  const remoteParticipants = useRemoteParticipants();
  ...
};
```

### Todos
- [x] Update `react-bindings`
- [x] Update `React SDK`
- [X] Update `React Native SDK`
- [X] Update React Sample Apps
- [X] Update React Native Sample Apps
- [ ] Update docs and Tutorials
- [ ] Test all apps

### CODE REVIEW

**Comment by khushal87** [comment]
@oliverlaz There are a couple of code smells as reported by Sonar cloud, can you have a look?

File: packages/react-bindings/src/hooks/index.ts [hunk 1]
```diff
@@ -1,5 +1,12 @@
-export * from './call';
-export * from './participants';
+import * as CallStateHooks from './callStateHooks';
+
 export * from './permissions';
 export * from './store';
-export * from './user';
+// export * from './callStateHooks';
```
Comment (khushal87) [inline packages/react-bindings/src/hooks/index.ts:5]: It seems that it can be removed

**Review by khushal87 — APPROVED** [review]
Approving the RN changes. Please look into the code smells reported by Sonarcloud before the merge. 😄

**Comment by oliverlaz** [comment]
> @oliverlaz There are a couple of code smells as reported by Sonar cloud that seems to be genuine, can you have a look?

They are a consequence of the newly deprecated `useCallMetadata()` hook. I'll remove that one next and eventually have these code smells removed.

File: packages/react-bindings/src/hooks/index.ts (hunk 1 above)
Comment (oliverlaz) [reply packages/react-bindings/src/hooks/index.ts]: Kept this export, in order to keep this PR backwards compatible. I'll remove this export once together with the removal of `useCallMetadata`.

**Review by MartinCupela — APPROVED** [review]

**Pull request merged**